sops.go

```diff
@@ -284,50 +287,64 @@ func (branch TreeBranch) walkBranch(in TreeBranch, path []string, onLeaves func(
-// If encryption is successful, it returns the MAC for the encrypted tree.
+// Encrypt walks over the tree and encrypts all values with the provided cipher,
+// except those whose key ends with the UnencryptedSuffix specified on the
+// Metadata struct, or those not ending with EncryptedSuffix, if EncryptedSuffix
+// is provided (by default it is not).  If encryption is successful, it returns
+// the MAC for the encrypted tree.
 func (tree Tree) Encrypt(key []byte, cipher Cipher) (string, error) {
 	audit.SubmitEvent(audit.EncryptEvent{
 		File: tree.FilePath,
 	})
 	hash := sha512.New()
-	_, err := tree.Branch.walkBranch(tree.Branch, make([]string, 0), func(in interface{}, path []string) (interface{}, error) {
-		// Only add to MAC if not a comment
-		if _, ok := in.(Comment); !ok {
-			bytes, err := ToBytes(in)
-			if err != nil {
-				return nil, fmt.Errorf("Could not convert %s to bytes: %s", in, err)
+	var err error
+	walk := func(branch TreeBranch) error {
+		_, err := branch.walkBranch(branch, make([]string, 0), func(in interface{}, path []string) (interface{}, error) {
+			// Only add to MAC if not a comment
+			if _, ok := in.(Comment); !ok {
+				bytes, err := ToBytes(in)
+				if err != nil {
+					return nil, fmt.Errorf("Could not convert %s to bytes: %s", in, err)
+				}
+				hash.Write(bytes)
 			}
-			hash.Write(bytes)
-		}
-		encrypted := true
-		if tree.Metadata.UnencryptedSuffix != "" {
-			for _, v := range path {
-				if strings.HasSuffix(v, tree.Metadata.UnencryptedSuffix) {
-					encrypted = false
-					break
+			encrypted := true
+			if tree.Metadata.UnencryptedSuffix != "" {
+				for _, v := range path {
+					if strings.HasSuffix(v, tree.Metadata.UnencryptedSuffix) {
+						encrypted = false
+						break
+					}
 				}
 			}
-		}
-		if tree.Metadata.EncryptedSuffix != "" {
-			encrypted = false
-			for _, v := range path {
-				if strings.HasSuffix(v, tree.Metadata.EncryptedSuffix) {
-					encrypted = true
-					break
+			if tree.Metadata.EncryptedSuffix != "" {
+				encrypted = false
+				for _, v := range path {
+					if strings.HasSuffix(v, tree.Metadata.EncryptedSuffix) {
+						encrypted = true
+						break
+					}
 				}
 			}
-		}
-		if encrypted {
-			var err error
-			pathString := strings.Join(path, ":") + ":"
-			in, err = cipher.Encrypt(in, key, pathString)
-			if err != nil {
-				return nil, fmt.Errorf("Could not encrypt value: %s", err)
+			if encrypted {
+				var err error
+				pathString := strings.Join(path, ":") + ":"
+				in, err = cipher.Encrypt(in, key, pathString)
+				if err != nil {
+					return nil, fmt.Errorf("Could not encrypt value: %s", err)
+				}
 			}
+			return in, nil
+		})
+		return err
+	}
+
+	for _, branch := range tree.Branches {
+		err = walk(branch)
+		if err != nil {
+			break
```

Comment:
We could just return the error here instead of breaking and checking the error again later.